internal/executor/executor.go

```diff
@@ -185,6 +185,16 @@ func getExpandedScriptEnvironment(executor *Executor, responseEnvironment map[st
 		} else if executor.commandFrom != "" {
 			// Default folder exists and we continue execution. Therefore we need to use it.
 			responseEnvironment["CIRRUS_WORKING_DIR"] = filepath.ToSlash(defaultTempDirPath)
+		} else if executor.cleanWorkingDir {
+			_ = os.RemoveAll(defaultTempDirPath)
+			// try to find random working dirs in case there was a cancellation on a persistent worker for example.
```

Comment:
It seems that this might remove working directories of still working agents.

What do you think about implementing this logic in the CLI's `cirrus worker` instead, as a task's finalizer? Should be cleaner in regards to responsibilities (we'll only remove directories that we've created).

Reply:
At the moment we don't officially documenting that it's safe to run multiple workers on a single host. It seems there might be a lot of such directories left on the existing workers.

Ideally yes, CLI should override a working dir for an agent and then make sure to cleaned it up after task cancelation.

Do you want to create a separate issue in CLI repo?

Reply:
>Ideally yes, CLI should override a working dir for an agent and then make sure to cleaned it up after task cancelation.

OK, will implement this in the CLI then 👌

Reply:
Sounds good. I guess the `cleanWorkingDir` feature can be removed from the agent again, now that the plan is to move it to the cli?

Reply:
> Sounds good. I guess the `cleanWorkingDir` feature can be removed from the agent again, now that the plan is to move it to the cli?

Exactly!